collector.go

```diff
@@ -348,17 +348,17 @@ func collectGenericSensor(ch chan<- prometheus.Metric, state float64, data senso
 	)
 }
 
-func (c collector) collectMonitoring(ch chan<- prometheus.Metric, creds Credentials) error {
+func (c collector) collectMonitoring(ch chan<- prometheus.Metric, creds Credentials) int {
```

Comment:
I would still keep this an `error` type. But for now make the return a `nil` for collector failures.

Reply:
I explicitly decided against this, for the same reason you mention below: the error is ignored, I don't see what the caller would do with. I guess the result would be three calls like `xUp, _ := collectX()`. I felt that seemed pointless, but open to suggestions.

Reply:
Yes, but the Go style is to use the `error` type. You can make decisions on specific errors, [like this](https://github.com/prometheus/node_exporter/blob/master/collector/zfs.go#L64-L65).

Reply:
@SuperQ Ok, I propagate the error now. I think this is sufficient for now. Error handling will be improved once you can select the collectors to run for each scrape, but at least this solves #1 for now.